src/goto-programs/goto_contractor.h

```diff
@@ -405,6 +405,23 @@ class CspMap
   {
     return is_empty_vector;
   }
+  void dump()
+  {
+    std::ostringstream oss;
+    oss << "This map has : " << this->n;
+    if (is_empty_vector)
+    {
+      oss << "empty vector";
+      log_status("{}", oss.str());
+      return;
+    }
+    log_status("{}", oss.str());
+
+    for (auto var : var_map)
```

Comment:
This will create a copy of every std::pair in the var_map. Please use `const auto &` or similar.

Reply:
done.